### PR TITLE
ECM-413 add judgement juridiction validation to case close event

### PR DESCRIPTION
**Pull request by Angela-LW**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ECM-413

### Change description ###
Add validation to stop cases that don't have a jurisdiction attached to a judgement from closing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java [hunk 1]
```diff
@@ -333,6 +334,24 @@ private void getJurisdictionCodesErrors(List<String> errors, List<String> jurCod
         return errors;
     }
 
+    public void validateJudgementsHasJurisdiction(CaseData caseData, boolean partOfMMultiple, List<String> errors) {
+        if (caseData.getJudgementCollection() == null || caseData.getJudgementCollection().isEmpty()) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java:338]: CollectionUtils.isNotEmpty covers both of these tests

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java [hunk 2]
```diff
@@ -333,6 +334,24 @@ private void getJurisdictionCodesErrors(List<String> errors, List<String> jurCod
...
+            return;
+        }
+
+        for (JudgementTypeItem judgementTypeItem : caseData.getJudgementCollection()) {
+            var judgementType = judgementTypeItem.getValue();
+            if (judgementType.getJurisdictionCodes() == null || judgementType.getJurisdictionCodes().isEmpty()) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java:344]: CollectionUtils again

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCloseEventValidationServiceTest.java [hunk 3]
```diff
@@ -60,22 +65,27 @@ public void multipleCloseEventValidationEmptyCaseIdCollection() {
...
         jurCodesTypeItem.setValue(new JurCodesType());
 
+        var judgmentTypeItem = new JudgementTypeItem();
+        judgmentTypeItem.setId("TEST");
+        judgmentTypeItem.setValue(new JudgementType());
+
         CaseData caseData = new CaseData();
         caseData.setEthosCaseReference("245004/2020");
         caseData.setJurCodesCollection(new ArrayList<>(Collections.singletonList(jurCodesTypeItem)));
+        caseData.setJudgementCollection(new ArrayList<>(Collections.singletonList(judgmentTypeItem)));
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCloseEventValidationServiceTest.java:88]: List.of maybe?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCloseEventValidationServiceTest.java [hunk 4]
```diff
@@ -94,27 +104,42 @@ public void multipleCloseEventValidationJurisdictionsCollectionReturnsErrors() {
...
+    @Test
+    public void multipleCloseEventValidationNoError() {
         var jurCodeType = new JurCodesType();
         jurCodeType.setJudgmentOutcome("some outcome");
         var jurCodesTypeItem = new JurCodesTypeItem();
         jurCodesTypeItem.setId("TEST");
         jurCodesTypeItem.setValue(jurCodeType);
 
+        var judgmentType = new JudgementType();
+        judgmentType.setJurisdictionCodes(new ArrayList<>(Collections.singletonList(jurCodesTypeItem)));
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCloseEventValidationServiceTest.java:134]: List.of

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java (hunk 1 above)
Comment (Angela-LW) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java]: updated ^_^

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java (hunk 2 above)
Comment (Angela-LW) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java]: updated ^_^

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCloseEventValidationServiceTest.java (hunk 4 above)
Comment (Angela-LW) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCloseEventValidationServiceTest.java]: updated ^_^

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCloseEventValidationServiceTest.java (hunk 3 above)
Comment (Angela-LW) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCloseEventValidationServiceTest.java]: updated ^_^

**Review by jthmcts — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]

**Review by btensay — APPROVED** [review]

**Pull request merged**